app/queries/servers/preference.ts

```diff
@@ -46,3 +47,15 @@ export const queryThemeForCurrentTeam = async (database: Database) => {
 
     return undefined;
 };
+
+export const deletePreferences = async (database: ServerDatabase, preferences: PreferenceType[]) => {
+    const preparedModels: Model[] = [];
+    for await (const pref of preferences) {
+        const myPrefs = await queryPreferencesByCategoryAndName(database.database, pref.category, pref.name);
+        for (const p of myPrefs) {
+            preparedModels.push(p.prepareDestroyPermanently());
+        }
+    }
+    await database.operator.batchRecords(preparedModels);
```

Comment:
Should be called only if there is anything to commit.

Aldo you should wrap in a try/catch

Reply:
I was debating on creating this function vs having another input parameter to `handlePreferences` and having the logic inside that function.  I thought i was better to have the separate function.